bundles/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/message/dto/LocationMessage.java

```diff
@@ -53,7 +53,7 @@ public class LocationMessage {
      * Altitude (iOS, Android/integer/meters/optional)
      */
     @SerializedName("alt")
-    private Integer altitude = Integer.MIN_VALUE;
+    private Float altitude = 0f;
```

Comment:
This breaks this check:
https://github.com/openhab/openhab-addons/blob/6115cd39a7298605735d416b3bc5c64f363d285b/bundles/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/message/dto/LocationMessage.java#L112

Secondly, why not `BigDecimal` like `latitude` and `longitude`?

Reply:
Because longitude needs atleast 5 decimals to have a 1 meter resolution. Altitude is allready in meters. There is no point in specifying a very exact altitude in micrometers.

Reply:
That makes sense. However, I'm mostly concerned about subtle rounding errors which can be avoided by using `BigDecimal` for precise decimal arithmetic, and is also commonly used in openHAB. I wouldn't expect memory or performance penalties here.

Reply:
I don't see a problem with using BigDecimal, so if that is more comfortable, i just changed it :-)